### PR TITLE
Add draft feature file: Filter Drawer Clear Filters

### PR DESCRIPTION
**Pull request by Bhavna-Ramachandran**
## Summary
Work for : Issue #[7075](https://github.com/HHS/simpler-grants-gov/issues/7075)

## Changes proposed
Created draft scenarios to verify and validate drawer 'Clear filters'
Added feature file: /frontend/tests/e2e/search/features/filterDrawer/filterDrawerClearFilters.feature

## Context for reviewers
Reviewers should verify that all critical user flows are covered.
The scenarios listed in the feature file will be automated using Playwright in future PRs.

## Validation steps
Confirm the feature file exists at: /frontend/tests/e2e/search/features/filterDrawer/filterDrawerClearFilters.feature
Review the Gherkin scenarios to ensure they cover all critical user flows.

### CODE REVIEW

File: frontend/tests/e2e/search/features/filterDrawer/filter-drawer-clear-filters.feature [hunk 1]
```diff
@@ -0,0 +1,86 @@
+Feature: Filter Drawer - Clear Filters
```
Comment (doug-s-nava) [inline frontend/tests/e2e/search/features/filterDrawer/filter-drawer-clear-filters.feature:1]: filename to `filter-drawer/filter-drawer-clear-filters`

File: frontend/tests/e2e/search/features/filterDrawer/filterDrawerClearFilters.feature [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    Then the filters should be cleared for "<loginState>"
+
+# Negative / Edge Cases
+Scenario: Clicking Clear Filters with no filters selected does nothing
+    Given no filters are selected
+    When I click the Clear Filters button
+    Then no filters should be selected
+    And results should remain unchanged
+
+Scenario: Clear Filters does not remove filters applied outside the drawer
```
Comment (doug-s-nava) [inline frontend/tests/e2e/search/features/filterDrawer/filterDrawerClearFilters.feature]: probably more clear to make this `Clear Filters does not remove an applied search term`, since the search term is not generally thought of as a filter. Could also add a separate case for the and / or switch

Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/search/features/filterDrawer/filterDrawerClearFilters.feature]: Acknowledged. Updates made.

File: frontend/tests/e2e/search/features/filterDrawer/filterDrawerClearFilters.feature [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+# Positive Scenarios
+Scenario: Clear Filters button is visible when filters are selected
+    Given I have selected one or more filters
+    Then the Clear Filters button should be visible
+
+Scenario: Clicking Clear Filters deselects all filters
+    Given I have selected one or more filters
+    When I click the Clear Filters button
+    Then all selected filters should be deselected
```
Comment (eric-serrato95) [inline frontend/tests/e2e/search/features/filterDrawer/filterDrawerClearFilters.feature]: @Bhavna-Ramachandran I think this might be a bit misleading, clear filters does not deselect everything but rather gets back to its default state, and by default there are 2 filters applied which is opportunities Posted and Forecasted

File: frontend/tests/e2e/search/features/filterDrawer/filter-drawer-clear-filters.feature (hunk 1 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/search/features/filterDrawer/filter-drawer-clear-filters.feature:1]: File name has been updated

File: frontend/tests/e2e/search/features/filterDrawer/filterDrawerClearFilters.feature (hunk 3 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/search/features/filterDrawer/filterDrawerClearFilters.feature]: Acknowledged @eric-serrato95 Below is the updated scenario:

Scenario: Clicking Clear Filters restores default filters
    Given the default filters are "Posted" and "Forecasted"
    And I have selected additional filters
    When I click the Clear Filters button
    Then only the default filters should remain selected

Comment (eric-serrato95) [reply frontend/tests/e2e/search/features/filterDrawer/filterDrawerClearFilters.feature]: correct @Bhavna-Ramachandran

**Review by eric-serrato95 — APPROVED** [review]
